test/gtest/uct/ib/test_ib.cc

```diff
@@ -259,27 +313,55 @@ class test_uct_ib : public uct_test {
 protected:
     entity *m_e1, *m_e2;
     static size_t ib_am_handler_counter;
+    struct ibv_device **device_list;
+    int num_devices;
 };
 
 size_t test_uct_ib::ib_am_handler_counter = 0;
 
-UCS_TEST_P(test_uct_ib, non_default_pkey, "IB_PKEY=0x2")
+UCS_TEST_P(test_uct_ib, non_default_pkey)
```

Comment:
maybe move all pkey testing to a new file, test_ib_pkey.cc?
it's getting pretty big..

Reply:
yes, this is a good idea - done